crates/madara/client/settlement_client/src/starknet/event.rs

```diff
@@ -269,108 +311,81 @@ mod starknet_event_stream_tests {
 
     #[tokio::test]
     #[rstest]
-    async fn test_single_event() {
-        let mock_server = MockStarknetServer::new();
-
-        // Setup mocks
-        let test_event = create_test_event(1, 100);
-        let events_mock = mock_server.mock_get_events(vec![test_event], None);
+    async fn test_single_event(mock_server: MockStarknetServer, test_event: EmittedEvent) {
+        let events_mock = mock_server.mock_get_events(vec![test_event.clone()], None);
         let block_mock = mock_server.mock_block_number(101);
 
-        let mut stream = Box::pin(setup_stream(&mock_server));
+        let mut stream = Box::pin(create_stream(&mock_server));
 
-        if let Some(Some(Ok(event_data))) = stream.next().await {
+        if let Some(Ok(event_data)) = stream.next().await {
             assert_eq!(event_data.block_number, 100);
             assert!(event_data.message_hash.is_some());
             assert_eq!(event_data.payload.len(), 2);
         } else {
             panic!("Expected successful event");
         }
 
-        // Verify mocks were called
-        events_mock.assert();
         events_mock.assert();
         block_mock.assert();
     }
 
     #[tokio::test]
     #[rstest]
-    async fn test_error_handling() {
-        let mock_server = MockStarknetServer::new();
-
-        let error_mock = mock_server.mock_error_response(-32000, "Internal error");
-
-        let mut stream = Box::pin(setup_stream(&mock_server));
-
-        match stream.next().await {
-            Some(Some(Err(e))) => {
-                assert!(e.to_string().contains("Internal error"));
-            }
-            _ => panic!("Expected error"),
-        }
+    async fn test_multiple_events_in_single_block(mock_server: MockStarknetServer) {
+        let event1 = test_event(1, 100);
+        let event2 = test_event(2, 100);
```

Comment:
This should be handled by `rstest` no? You can generate multiple instance of the same fixture with `#[from(your_fixture)]`. You can learn more about this in the [injecting fixtures](https://docs.rs/rstest/0.24.0/rstest/attr.rstest.html#injecting-fixtures) section of the `rstest` docs.
```suggestion
    async fn test_multiple_events_in_single_block(
        mock_server: MockStarknetServer,
        #[from(test_event)] event1: EmittedEvent,
        #[from(test_event)] event2: EmittedEvent,        
    ) {
```